crypto/store/store_init.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2001-2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/err.h>
+#include "internal/store.h"
+#include "store_local.h"
+
+static CRYPTO_ONCE store_init = CRYPTO_ONCE_STATIC_INIT;
+DEFINE_RUN_ONCE_STATIC(do_store_init)
+{
+    return OPENSSL_init_crypto(0, NULL)
```

Comment:
Yes, this is necessary.  Take a very simple program:

``` C
#include <openssl/store.h>

int main()
{
  STORE_CTX *store = STORE_open("file:/foo/bar.pem");

  if (store == NULL) {
    ERR_print_errors_dp(stderr);
    exit(1);
  ...
}
```

A perfectly legitimate start, and this ends up calling `store_get0_loader_int` with the scheme name, which in turn calls `store_init_once`, and that initialises the crypto library, including all the error stuff that's needed.

So yes, this is needed.

Is it likely that something in `OPENSSL_init_crypto()` tries to load a store?  I've my doubts, but...

Oh, and the other two that call `store_init_once` are `STORE_register_loader` and `STORE_unregister_loader`.  I think that should cover all likely times this kind of init is needed.